### DISCUSSION TITLE
Feature Request: output filename.csv generated from sql variable

### DISCUSSION THREAD

**Discussion by jimdickson007** [duckdb/duckdb#14006] [Ideas]
I can read csv using variable filename
I can write csv using fixed filename
Can I write csv using variable filename ?

Currently using COPY but happy if alternative exists.

**Comment by frankzandhuis** [comment]
Same question from my side, but then writing to parquet.

I set a variable with a path that I want to use to read/ write parquet files. I can read parquets based on a variable with the read_parquet() function. However, I'm unable to write to a parquet file using the COPY TO function with a variable.

```
SET VARIABLE myPath = 'C:\path\to\file';
```

This works:

```
SELECT * FROM read_parquet(getvariable('myPath') || ("\duckdb\myFile.parquet"));
```

This also works, but I would prefer to use the variable here as well:


```
) TO "C:\path\to\file\duckdb\myFile.parquet" (FORMAT 'parquet');
```

**Comment by jimdickson007** [comment]
I am guessing this is not possible ?
Not sure what site etiquette - do I close ? If so, which option is most appropriate / helpful to community ?

![image](https://github.com/user-attachments/assets/ef652b2e-0214-4ec5-aff8-acd12815ec6f)

**Comment by carlopi** [comment]
This is not possible at the moment, as far as I know. This it's a valid feature request, maybe can you change the title like "Feature: Support variable as output file name" or something to make it more clear / attract visibility ?

**Comment by Mickychen00** [comment]
Hi, is there a solution now?

**Comment by Mytherin** [comment]
This has been implemented in https://github.com/duckdb/duckdb/pull/18795
